app_widgetbook/lib/core/ui_elements/app_elevated_button.dart

```diff
@@ -0,0 +1,167 @@
+import 'package:app_widgetbook/utils.dart';
+import 'package:core/core.dart';
+import 'package:flutter/material.dart';
+import 'package:widgetbook/widgetbook.dart' show Knobs, Option;
+import 'package:widgetbook_annotation/widgetbook_annotation.dart';
+
+/// Use case for a Primary Button style
+@WidgetbookUseCase(name: 'Custom', type: AppElevatedButton)
+Widget customAppElevatedButtonUseCase(BuildContext context) {
+  return AppElevatedButton(
+    label: context.knobs.nullableText(
+      label: 'Label',
+      initialValue: 'Button',
+    ),
+    icon: context.knobs.boolean(label: 'Has Icon', initialValue: true)
+        ? context.knobs.options(
+            label: 'Icon',
+            options: buttonIconOptions,
+          )
+        : null,
+    onPressed: () {},
+    borderColor: context.knobs.options<Color?>(
+      label: 'Border Color',
+      options: const [
+        Option(
+          label: 'None',
+          value: null,
+        ),
+        Option(
+          label: 'Primary',
+          value: AppColors.primary,
+        ),
+        Option(
+          label: 'Secondary',
+          value: AppColors.secondary,
+        ),
+        Option(
+          label: 'Pink',
+          value: AppColors.pink,
+        ),
+        Option(
+          label: 'White',
+          value: AppColors.white,
+        ),
+      ],
+    ),
+    backgroundColor: context.knobs.options<Color?>(
+      label: 'Background Color',
+      options: const [
+        Option(
+          label: 'Primary',
+          value: AppColors.primary,
+        ),
+        Option(
+          label: 'Secondary',
+          value: AppColors.secondary,
+        ),
+        Option(
+          label: 'Pink',
+          value: AppColors.pink,
+        ),
+        Option(
+          label: 'White',
+          value: AppColors.white,
+        ),
+      ],
+    ),
+    textColor: context.knobs.options<Color?>(
+      label: 'Text Color',
+      options: const [
+        Option(
+          label: 'White',
+          value: AppColors.white,
+        ),
+        Option(
+          label: 'Primary',
+          value: AppColors.primary,
+        ),
+        Option(
+          label: 'Secondary',
+          value: AppColors.secondary,
+        ),
+        Option(
+          label: 'Pink',
+          value: AppColors.pink,
+        ),
+      ],
+    ),
+    height: context.knobs.boolean(label: 'Is Large') ? 56 : 40,
+  );
+}
+
+/// Use case for a Primary Button style
+@WidgetbookUseCase(name: 'Primary Button', type: AppElevatedButton)
+Widget primaryAppElevatedButtonUseCase(BuildContext context) {
+  return AppElevatedButton.primary(
+    label: context.knobs.nullableText(
+      label: 'Label',
+      initialValue: 'Button',
+    ),
+    icon: context.knobs.boolean(label: 'Has Icon', initialValue: true)
+        ? context.knobs.options(
```

Comment:
@jenshor  Just FYI, this is an example of where a `nullableOptions` knob would be useful similar to the `nullableText` above it. And this is already added in the features ClickUp doc 👍

Reply:
Agreed! Some users also wished for knobs that can depend on each other. E.g. a boolean knob that dis/enables another knobs. Not sure how easy it is to implement, tho. However, nullable option knob should not be a problem. 
However, you could also make. a `Option<T?>` instead of having `Option<T>` and then have a null value. 🤔